biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -563,6 +607,38 @@ public void annotation(Annotation a) {
 								// Ignore
 							}
 						}
+						break;
+					case "org.osgi.annotation.bundle.Export" :
+						Object[] attributes = a.get("attribute");
+
+						if (attributes != null) {
+							for (Object s : attributes) {
+								String[] attr = ((String) s).split("=", 2);
+								if (attr.length != 2) {
+									warning("The annotation attribute %s of the exported package %s could not be understood",
+											s, clazz.getClassName().getPackageRef().getFQN());
+									continue;
+								}
+								info.put(attr[0], attr[1]);
+							}
+						}
+
+						Object[] usesClause = a.get("uses");
+						if (usesClause != null) {
+							String old = info.get(USES_DIRECTIVE);
+							if (old == null)
+								old = "";
+							StringBuilder sb = new StringBuilder(old);
+							String del = sb.length() == 0 ? "" : ",";
+
+							for (Object use : usesClause) {
+								sb.append(del);
+								sb.append(use);
+								del = ",";
+							}
+							info.put(USES_DIRECTIVE, sb.toString());
+						}
+
 						break;
 					case "aQute.bnd.annotation.Export" :
```

Comment:
I would put the warning for using the Bnd Export annotation here rather than a separate stream check like above.

Reply:
This code is called to scan every entry on the class path - I originally tried putting the warning here but it caused bnd to complain about classpath libraries that were nothing to do with the jar being built. 

Due to the order in which things are done by bnd I was unable to find a way to determine whether the package is contained at the time of doing this scan (we only update the contained Packages after calling this method). It seems preferable to only warn for classes included in the bundle.

Reply:
Fair enough!